system/include/emscripten/wasmfs.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <sys/stat.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct Backend* backend_t;
+
+// Obtains backend_t of a specified path.
+backend_t wasmfs_get_backend(char* path);
+
+// Creates a JSFile Backend in the new file system.
+backend_t create_js_file_backend();
+
+// Creates a new file in the new file system under a specific backend.
+uint32_t wasmfs_create(char* pathname, mode_t mode, backend_t backend);
```

Comment:
`create` for data files and `mkdir` for directories doesn't seem that consistent to me. mkdir is a syscall name, but create isn't, in particular. but more generally, I'm not sure if we want to mirror the syscall names here or not, to be honest. Sorry I don't have a better suggestion.

Reply:
How about `wasmfs_create_file` and `wasmfs_create_directory`, since we are diverging from any related POSIX definitions and implementing our own functionality.